django_prices/models.py

```diff
@@ -146,6 +50,112 @@ def contribute_to_class(self, cls, name, **kwargs):
+        self.currency_field = currency_field
+
+    def __str__(self):
+        return "MoneyField(amount_field=%s, currency_field=%s)" % (
+            self.amount_field,
+            self.currency_field,
+        )
+
+    def __get__(self, instance, cls=None):
+        if instance is None:
+            return self
+
+        amount = getattr(instance, self.amount_field)
+        currency = getattr(instance, self.currency_field)
+        if amount and currency:
+            return Money(amount, currency)
+        return self.get_default()
+
+    def __set__(self, instance, value):
+        amount = None
+        currency = None
+        if value is not None:
+            amount = value.amount
+            currency = value.currency
+        setattr(instance, self.amount_field, amount)
+        setattr(instance, self.currency_field, currency)
+
+    def formfield(self, **kwargs):
+        available_currencies = []
+        if hasattr(self, "model"):
+            available_currencies = self.model._meta.get_field(
+                self.currency_field
+            ).choices
+        return forms.MoneyField(available_currencies=available_currencies)
+
+    def get_default(self):
+        default_currency = None
+        default_amount = Decimal(0)
+        if hasattr(self, "model"):
+            default_currency = self.model._meta.get_field(
+                self.currency_field
+            ).get_default()
+            default_amount = self.model._meta.get_field(self.amount_field).get_default()
+
+        return Money(default_amount, default_currency)
+
+
+class TaxedMoneyField(NonDatabaseFieldBase):
+
+    description = "A field that combines net and gross fields values into TaxedMoney."
+
+    def __init__(
+        self,
+        net_amount_field="price_amount_net",
+        gross_amount_field="price_amount_gross",
+        currency="currency",
+        verbose_name=None,
+        **kwargs
+    ):
+        super(TaxedMoneyField, self).__init__()
+        self.net_amount_field = net_amount_field
+        self.gross_amount_field = gross_amount_field
+        self.currency = currency
+
+    def __str__(self):
+        return (
+            "TaxedMoneyField(net_amount_field=%s, gross_amount_field=%s, currency=%s)"
+            % (self.net_amount_field, self.gross_amount_field, self.currency)
+        )
+
+    def __get__(self, instance, cls=None):
+        if instance is None:
+            return self
+        net_amount = getattr(instance, self.net_amount_field)
+        gross_amount = getattr(instance, self.gross_amount_field)
+        currency = getattr(instance, self.currency)
+        return TaxedMoney(Money(net_amount, currency), Money(gross_amount, currency))
+
+    def __set__(self, instance, value):
+        net_amount = None
+        gross_amount = None
+        currency = None
+        if value is not None:
+            net_amount = value.net.amount
+            gross_amount = value.gross.amount
+            currency = value.gross.currency
```

Comment:
I think you can also use `value.currency` here

Reply:
You're right, I've changed it for better readability.